### PR TITLE
IGVF-309-table-cell-renders

### PR DESCRIPTION
**Pull request by forresttanaka**
I broke my own rule very badly in this branch. Many Jest tests now fail coveralls coverage because the amount of time it will take to fix all these at this stage would be prohibitive IMO. I would like to have IGVF-324 as a high priority, maybe before IGVF-318.

### CODE REVIEW

File: pages/genes/[id].js [hunk 1]
```diff
@@ -78,18 +79,9 @@ const Gene = ({ gene }) => {
             )}
             {gene.locations?.length > 0 && (
               <>
-                <DataItemLabel>Gene Locations</DataItemLabel>
+                <DataItemLabel>Gene GeneLocations</DataItemLabel>
```
Comment (mingjiecn) [inline pages/genes/[id].js:82]: Why change "Gene Locations" to "Gene GeneLocations"?

Comment (forresttanaka) [reply pages/genes/[id].js:82]: That’s a good question. Must have been a stray paste error.

**Comment by mingjiecn** [comment]
How is the row length determined? Some rows has more content in certain cells but the rows are narrower compare to other rows. For example: external resources column is very narrow.
image

**Comment by forresttanaka** [comment]
Column widths are determined by the browser. We can adjust it, and Ben and I talked about it this morning as something to work on in the future. But right now every column has the width the browser’s algorithm wants it.

**Comment by mingjiecn** [comment]
OK. Then I am good after "Gene Locations" issue is fixed.

**Review by mingjiecn — CHANGES_REQUESTED** [review]
fix paste error for "Gene Locations"

**Review by mingjiecn — APPROVED** [review]

**Comment by forresttanaka** [comment]
@mingjiecn The package-lock.json file simply needs rebuilding to resolve the conflict, which I’ll do before merging to dev.

**Comment by forresttanaka** [comment]
## Note for second round of code review:

* I didn’t take the downloaded TSV file into account with this change, so many of the cell contents in the TSV file no longer (or never) made sense, or didn’t take advantage of the new mechanisms. This change fixes that by using a similar but much simpler mechanism to have custom renderers for cells in the TSV file, though since they only output text and not React components, I call them “text composers” here instead of “renderers.”
* The non-breaking spaces I put into date displays in the first PR caused more problems than it solved when it came to TSV generation. I removed all of that.

**Comment by mingjiecn** [comment]
NIH Institutional Certification is shown differently in list and download:
image
image

**Review by mingjiecn — CHANGES_REQUESTED** [review]
NIH Institutional Certification is shown differently in list and download.

**Comment by forresttanaka** [comment]
That appears to be a bug in the Numbers app. If you type “NIH00012" into a cell, it appears as “NIC 12.00.”

In Excel, it appears correctly. I don’t think there’s anything I can do about Numbers:
image

**Comment by mingjiecn** [comment]
You are right!

**Review by mingjiecn — DISMISSED** [review]

**Pull request merged**